src/ESPEasyRules.ino

```diff
@@ -156,42 +156,83 @@ String rulesProcessingFile(const String& fileName, String& event) {
   fs::File f = tryOpenFile(fileName, "r+");
   SPIFFS_CHECK(f, fileName.c_str());
 
-  String line      = "";
-  bool   match     = false;
-  bool   codeBlock = false;
-  bool   isCommand = false;
-  bool   condition[RULES_IF_MAX_NESTING_LEVEL];
-  bool   ifBranche[RULES_IF_MAX_NESTING_LEVEL];
-  byte   ifBlock     = 0;
-  byte   fakeIfBlock = 0;
-
-  byte *buf = new byte[RULES_BUFFER_SIZE]();
+  String line;
+  line.reserve(RULES_IF_MAX_NESTING_LEVEL);
+  bool match     = false;
+  bool codeBlock = false;
+  bool isCommand = false;
+  bool condition[RULES_IF_MAX_NESTING_LEVEL];
+  bool ifBranche[RULES_IF_MAX_NESTING_LEVEL];
+  byte ifBlock     = 0;
+  byte fakeIfBlock = 0;
+
+  std::vector<byte> buf;
+  buf.resize(RULES_BUFFER_SIZE);
+
+  bool firstNonSpaceRead = false;
+  bool commentFound      = false;
 
   while (f.available()) {
-    int len = f.read((byte *)buf, RULES_BUFFER_SIZE);
+    int len = f.read(&buf[0], RULES_BUFFER_SIZE);
 
     for (int x = 0; x < len; x++) {
       int data = buf[x];
 
       SPIFFS_CHECK(data >= 0, fileName.c_str());
 
-      if (data != 10) {
-        line += char(data);
-      } else { // if line complete, parse this rule
-        line.replace("\r", "");
+      switch (static_cast<char>(data))
+      {
+        case 10: // "\n"
```

Comment:
just case '\n'

Reply:
True, that was a left-over from when the compiler said "no" this morning :)

Reply:
same for 32  - ' '

Reply:
See https://github.com/letscontrolit/ESPEasy/pull/2617/commits/b44abc01936e815db8e4c90e099a6315ac51465e